mifosng-android/src/main/java/com/mifos/utils/CheckSelfPermissionAndRequest.java

```diff
@@ -128,8 +131,16 @@ public void onClick(DialogInterface dialog, int which) {
                                                 R.string.package_name), activity.getPackageName()
                                                 , null);
                                         intent.setData(uri);
-                                        activity.startActivityForResult(intent,
-                                                Constants.REQUEST_PERMISSION_SETTING);
+                                        try {
+                                            activity.startActivityForResult(intent,
+                                                    Constants.REQUEST_PERMISSION_SETTING);
+                                        } catch (ActivityNotFoundException exception) {
+                                            Toast.makeText(activity,
+                                                    R.string.app_settings_activity_failed
+                                                    , Toast.LENGTH_SHORT)
+                                                 .show();
+                                            Log.d("MifosApp", "cant find activity with name");
```

Comment:
Good Catch @devkhan 

TAG should be a constant member name and also meaningful! 
MifosApp as a tag is bad choice for debugging purposes :)